### ISSUE TITLE
Milestone 2 To Do List & Relevant Discussion

### ISSUE THREAD

**Issue by jnederlo** [UBC-MDS/DSCI_522_group408#17]
- [x] Script to download data and save it locally (already done but might need to change to R). **- Roc**
  - [x] Tests on script.
- [x] Script to read data from the first script and perform wrangling. **- Roc**
  - [x] Tests on script.
- [x] Script to do the EDA. **- Roc**
  - [x] Tests on Script.
  - discussion notes:  correlation chart / table, histogram of the target , ewm features
- [x] Script to run the ML analysis. **- Jack**
  - [x] Tests on Script.
  - discussion notes:  
    1. models to try: benchmark model (last 5 avg), XGBoost, LGBM, RF, some Regression, and maybe more; 
    2. report RMSE (maybe also R^2) for all the models (and perhaps also the correlation between y_pred and y_true?)
    3. for the final model: 
       1. report feature importance
       2. make bin scatter plots of y-true vs y-pred and benchmark
       3. residual plots? or any other reasonable plots
- [x] Final report in Markdown or Jupyter. **- Jarvis**
- [x] Project References (citations). **- Jarvis**
- [x] Create a new release (version 0.2.0).
- [x] Update the README.

**Comment by jacktan1** [comment]
Figures after creating models:
- [x] Residual vs predicted values for all models.
- [x] XGBoost
- [x] Linear/Logistic Regression
- [x] Loess (maybe)
- [x] Mean squared error loss function
- [x] LGBM
EDA Analysis:
- [x] Distribution of minutes played
- [x] Correlation charts/table

**Comment by Zhang-Haipeng** [comment]
I've just updated our discussion note in the first comment. Please take a look and see if there are any missing points.  
  
Another thing, I'd suggest naming our scripts starting with numbers like:  
00-download_data  
01-data_preprocessing  
02-EDA  
03-...

**Comment by Zhang-Haipeng** [comment]
Hey guys. I've just created a PR splitting data_preprocessing and EDA into two files. Also, I've added `ewm` features with `alpha` = [0.1, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7].   
While I continue my part of work, @thejacktan could use this as the input for the modeling part.   
Also, it's worth mentioning that `ewm` features do beat previous regular rolling features in correlation with the response. So we might want to take a look at the important features of the model, to decide whether to involve so many rolling features at all.

**Comment by Zhang-Haipeng** [comment]
@thejacktan Jarvis and I did some feature selection work during the lab so that our data pre-processing won't take too much time to reproduce and our model won't be "overwhelmed" by too many weak features.  Please make sure to merge the latest upstream and rerun `02-data_preproc.ipynb` before exploring the modeling part.

**Comment by jnederlo** [comment]
@thejacktan After you load the dataset you will need to remove some columns that we don't want to train on but we do want to keep for reference. Therefore, after loading, run this block of code:

`info_cols = ['playDispNm', 'gmDate', 'teamAbbr', 'playPos']`
`info_df = df[info_cols]`
`df.drop(columns=info_cols, inplace=True)`

The first line will identify columns that we don't want to use in our model (i.e. these are ID columns),
the second line will create a subset of the original dataframe with those columns, and the third line will remove those columns from the original dataframe so we don't train on them.

After training is done, we will need to merge this information back to the players so we can show some insights. I've given some code examples to get the predictions back into this `info` dataframe:

#### Make predictions with the model
`y_pred = model.predict(X)`

#### Make a dataframe to hold the predictions
`preds = pd.DataFrame()`
`preds['Actual'] = Y`
`preds['Predicted'] = y_pred`
`preds.sort_values(by='Actual', inplace=True)`

#### Add the predictions and actual minutes played to the info dataframe
`info['pred_min'] = preds['Predicted']`
`info['actual_min'] = preds['Actual']`
`info.sort_values(by='Actual', inplace=True)`

Once you get the `info` dataframe then we can go about making a table of some of the interesting results, etc. The figures we create from this dataframe will go in our final report.

**Comment by Zhang-Haipeng** [comment]
@jnederlo I've made a PR containing the outputs of EDA for the reference of the report.    
And @thejacktan let's put all the outputs into the `results` folder, naming them `EDA-xxx`, `Modeling-xxx` (or whatever prefixes you prefer).

**Comment by jnederlo** [comment]
Milestone 2 Tests finished and complete with PR #35
